Source/WTF/wtf/text/StringCommon.h

```diff
@@ -967,16 +967,24 @@ inline bool equalIgnoringASCIICase(ASCIILiteral a, ASCIILiteral b)
 }
 
 template<typename ElementType>
-inline void copyElements(ElementType* __restrict destination, const ElementType* __restrict source, size_t length)
+inline void copyElements(std::span<ElementType> destinationSpan, std::span<const ElementType> sourceSpan)
 {
-    if (length == 1)
+    ASSERT(destinationSpan.size() >= sourceSpan.size());
+    auto* __restrict destination = destinationSpan.data();
+    auto* __restrict source = sourceSpan.data();
```

Comment:
It's a bit suspicious that this code uses `__restrict` yet it doesn't verify that the memory regions don't overlap.
Can we add an assertion to that end?

Reply:
Not new to my patch but I can add such assertion, yes

Reply:
Done.